client/src/featureform/resources.py

```diff
@@ -667,19 +666,22 @@ def __post_init__(self):
         for key, val in self.properties.items():
             self.serialized.property[key].string_value = val
 
+
 @typechecked
 @dataclass
 class Provider:
     name: str
-    function: str
-    config: Config
     description: str
     team: str
-    tags: list
-    properties: dict
+    config: Optional[Config] = None  # TODO remove optional and add deserializer to configs
```

Comment:
Can we resolve this TODO easily, I don't like adding TODOs in code because they tend to never get done

Reply:
We can do it early next week, not worth blocking the release on obviously but lets knock it out soon